code/modules/projectiles/guns/rifles.dm

```diff
@@ -643,6 +643,143 @@
+
+
+/obj/item/weapon/gun/rifle/xm52/set_gun_config_values()
+	..()
+	fire_delay = FIRE_DELAY_TIER_5*2
+	burst_amount = 0
+	accuracy_mult = BASE_ACCURACY_MULT + HIT_ACCURACY_MULT_TIER_3
+	accuracy_mult_unwielded = BASE_ACCURACY_MULT - HIT_ACCURACY_MULT_TIER_10
+	scatter = SCATTER_AMOUNT_TIER_6
+	scatter_unwielded = SCATTER_AMOUNT_TIER_2
+	damage_mult = BASE_BULLET_DAMAGE_MULT
+	recoil = RECOIL_AMOUNT_TIER_4
+	recoil_unwielded = RECOIL_AMOUNT_TIER_2
+
+/obj/item/weapon/gun/rifle/xm52/able_to_fire(mob/user)
+	. = ..()
+	if(is_locked && linked_human && linked_human != user)
+		if(linked_human.is_revivable() || linked_human.stat != DEAD)
+			to_chat(user, SPAN_WARNING("[icon2html(src, usr)] Trigger locked by [src]. Unauthorized user."))
+			playsound(loc,'sound/weapons/gun_empty.ogg', 25, 1)
+			return FALSE
+
+		linked_human = null
+		is_locked = FALSE
+		UnregisterSignal(linked_human, COMSIG_PARENT_QDELETING)
+
+/obj/item/weapon/gun/rifle/xm52/pickup(user)
+	if(!linked_human)
+		name_after_co(user)
+		to_chat(usr, SPAN_NOTICE("[icon2html(src, usr)] You pick up \the [src], registering yourself as its owner."))
+	..()
+
+//---ability actions--\\
+
+/datum/action/item_action/xm52/action_activate()
+	var/obj/item/weapon/gun/rifle/xm52/protag_gun = holder_item
+	if(!ishuman(owner))
+		return
+	var/mob/living/carbon/human/protagonist = owner
+	if(protagonist.is_mob_incapacitated() || protag_gun.get_active_firearm(protagonist, FALSE) != holder_item)
+		return
+
+/datum/action/item_action/xm52/update_button_icon()
+	return
+
+/datum/action/item_action/xm52/toggle_id_lock/New(Target, obj/item/holder)
+	. = ..()
+	name = "Toggle ID lock"
+	action_icon_state = "id_lock_locked"
+	button.name = name
+	button.overlays.Cut()
+	button.overlays += image('icons/mob/hud/actions.dmi', button, action_icon_state)
+
+/datum/action/item_action/xm52/toggle_id_lock/action_activate()
+	. = ..()
+	var/obj/item/weapon/gun/rifle/xm52/protag_gun = holder_item
+	protag_gun.toggle_lock()
+	if(protag_gun.is_locked)
+		action_icon_state = "id_lock_locked"
+	else
+		action_icon_state = "id_lock_unlocked"
+	button.overlays.Cut()
+	button.overlays += image('icons/mob/hud/actions.dmi', button, action_icon_state)
+
+
+// -- ability actions procs -- \\
+
+/obj/item/weapon/gun/rifle/xm52/proc/toggle_lock(mob/user)
+	if(linked_human && usr != linked_human)
+		to_chat(usr, SPAN_WARNING("[icon2html(src, usr)] Action denied by [src]. Unauthorized user."))
+		return
+	else if(!linked_human)
+		name_after_co(usr)
+
+	is_locked = !is_locked
+	to_chat(usr, SPAN_NOTICE("[icon2html(src, usr)] You [is_locked? "lock": "unlock"] [src]."))
+	playsound(loc,'sound/machines/click.ogg', 25, 1)
+
+/obj/item/weapon/gun/rifle/xm52/proc/name_after_co(mob/living/carbon/human/H)
+	linked_human = H
+	RegisterSignal(linked_human, COMSIG_PARENT_QDELETING, PROC_REF(remove_idlock))
```

Comment:
```suggestion
/obj/item/weapon/gun/rifle/xm52/proc/register_owner(mob/living/carbon/human/H)
	linked_human = H
	to_chat(H, SPAN_NOTICE("[icon2html(src, usr)] You place your hand on \the [src], and register your biometrics."))
	RegisterSignal(linked_human, COMSIG_PARENT_QDELETING, PROC_REF(remove_idlock))
```
Should use proc name that is more indicative of what the proc actually does.
Also makes more sense for an update message to be in the update proc, rather than separate for each place proc is called.